engine/src/test/java/org/hibernate/search/test/backend/lucene/ScheduledCommitPolicyTest.java

```diff
@@ -80,6 +80,26 @@ public boolean evaluate() {
 	}
 
 	@Test
+	@BMRule(targetClass = "org.hibernate.search.backend.impl.lucene.IndexWriterHolder",
+			targetMethod = "commitIndexWriter()",
+			action = "throw new NullPointerException(\"Fake internal error\")",
+			name = "timerDisruptingError")
+	public void testErrorHandlingOnBackgroundThread() throws Exception {
+		writeData( sfAsyncExclusiveIndex, 2 );
+		final CountingErrorHandler errorHandler = (CountingErrorHandler) sfAsyncExclusiveIndex.getSearchFactory().getErrorHandler();
+		assertConditionMet( new Condition() {
+			@Override
+			public boolean evaluate() {
+				// It's going to commit once each millisecond, and produce a failure each time.
+				// So "4" is just a random number higher than 0, but high enough to
```

Comment:
that's a rebase/conflict mistake. I actually meant to use "4" in the code, which is safer than 2 as t here are 2 updates being done, I want to verify the scheduler does more than that.